### PR TITLE
addListener on interpolated values

### PR DESCRIPTION
**Pull request by browniefed**
Always wanted listeners on interpolated values. 
Not super sure if I just introduced a memory leak by adding a listener to the parent and removing it in detach?

### CODE REVIEW

**Comment by browniefed** [comment]
This is more of a PR as a question :). Not sure how to implement this correctly.

File: Animated.js [hunk 1]
```diff
@@ -682,11 +682,19 @@ class AnimatedVec2 extends AnimatedWithChildren {
...
   _interpolation: (input: number) => number | string;
+  _listeners: {[key: number]: ValueListenerCallback};
+  _parentListener: number;
 
   constructor(parent: Animated, interpolation: (input: number) => number | string) {
     super();
     this._parent = parent;
     this._interpolation = interpolation;
+    this._listeners = {};
+    this._parentListener = parent.addListener(() => {
```
Comment (vjeux) [inline Animated.js]: most of the time, there's --not-- going to be a listener here. So this is going to be very expensive to setup all those listeners. I would do it on the fly when the first addListener is being called

**Comment by browniefed** [comment]
Moved it

File: index.html [hunk 2]
```diff
@@ -324,15 +324,20 @@
       anim: new Animated.Value(1),
     };
   },
+  componentWillMount: function() {
+    this.interpolate = this.state.anim.interpolate({
```
Comment (vjeux) [inline index.html:328]: indent is off

File: index.html [hunk 3]
```diff
@@ -324,15 +324,20 @@
...
+              inputRange: [0, 1],
+              outputRange: ['260deg', '0deg']
+            })
```
Comment (vjeux) [inline index.html]: missing ;

File: index.html [hunk 4]
```diff
@@ -324,15 +324,20 @@
...
+
+    this.interpolate.addListener(v => console.log(v));
```
Comment (vjeux) [inline index.html:333]: :+1: 

The api is really nice :)

**Comment by vjeux** [comment]
I like it :)

**Comment by vjeux** [comment]
Feel free to merge whenever you want

**Comment by brentvatne** [comment]
nice @browniefed :dancer:

**Pull request merged**